File: dashboard/frontend/src/types/history.ts
export interface HistoryEntry {
  ticker: string
  date: string
  rating?: string
  raw_return?: string
  alpha_return?: string
  holding_days?: string
  pending: boolean
}

export interface HistoryListResponse {
  entries: HistoryEntry[]
  total: number
  limit: number
  offset: number
}

export interface TickerSummary {
  ticker: string
  total_analyses: number
  latest_date?: string
  latest_rating?: string
}

File: dashboard/api/schemas/history.py
"""Pydantic schemas for history browsing."""

from __future__ import annotations

from typing import Optional
from pydantic import BaseModel


class HistoryEntry(BaseModel):
    ticker: str
    date: str
    rating: Optional[str] = None
    raw_return: Optional[str] = None
    alpha_return: Optional[str] = None
    holding_days: Optional[str] = None
    pending: bool = False


class HistoryListResponse(BaseModel):
    entries: list[HistoryEntry]
    total: int
    limit: int
    offset: int


class TickerSummary(BaseModel):
    ticker: str
    total_analyses: int
    latest_date: Optional[str] = None
    latest_rating: Optional[str] = None


class CompareRequest(BaseModel):
    tickers: list[str]
    dates: Optional[list[str]] = None
